docs/dcp2_system_design.rst

```diff
@@ -684,12 +684,54 @@ Object names given in this section are relative to the staging area. To
 produce the complete ``gs://…`` URI of a particular object in the
 staging area, append the object's name to the staging area URI. [#]_
 
+
+Staging area properties
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Every staging area must include an object named ``staging_area.json`` in the
+root directory. It contains a JSON object that defines the *staging area
+properties* i.e. traits that apply to the staging area as a whole. Absence of
+this object must result in an immediate abort of the importer with an error
+message as defined in `Import errors`_. Legacy staging areas must be
+retrofitted with this object in order to be imported again.
+
+|nn| In this case, retrofitting seemed easier than defining defaults for each
+and every staging area property going forward. Similarly, as we add staging
+area  properties in the future, legacy staging areas would need to be
+retrofitted again with those new properties, but I feel it's a good thing that
+staging area sources are forced to make a concsious choice about the value of
+each new staging area property before their staging area can be imported
+again. |ne|
+
+The contents of ``staging_area.json`` must match the following schema::
+
+   {
+     "$schema": "https://json-schema.org/draft/2019-09/schema",
+     "properties": {
+       "is_delta": {
+         "type": "boolean"
+       }
+     },
+     "required": [
+       "is_delta"
+     ],
+     "additionalProperties": false
+   }
+
+The ``is_delta`` property indicates whether a staging only contains altered
+(added, deleted or updated) meta(data). The specifics are defined in `Types of
```

Comment:
```suggestion
(deleted) meta(data). The specifics are defined in `Types of
```

Reply:
This would change the semantics too much. If delta meant "deletions are present", the name would be wrong. The emphasis is on *"only"* here. Delta=true really means that there cannot be any unchanged entities in the staging area. Anything in the staging area must be genuinely different from what was last imported. The fact that a non-delta staging area can contain updated entities in addition to unchanged ones is only for backwards compatibility with how we are already doing things. For example, the analysis staging areas by Jessica and Gordon already exercised that semantics so I can't write that out of the spec without breaking their existing code.

Reply:
But I realize now that the text is inconsistent in this respect. Elsewhere it says that delta staging areas *should* not contain unchanged stuff, as opposed to *must not*. I will change that because it will be less confusing.

Reply:
Addressed. Didn't adopt the change but laid out my reasoning, clarified the text and removed the inconsistent portion.